src/openrct2-ui/TextComposition.cpp

```diff
@@ -36,19 +36,16 @@ bool TextComposition::IsActive()
     return SDL_IsTextInputActive() && _session.Buffer != nullptr;
 }
 
-TextInputSession* TextComposition::Start(utf8* buffer, size_t bufferSize)
+TextInputSession* TextComposition::Start(u8string& buffer, size_t maxLength)
```

Comment:
Can't we just own the buffer in TextComposition

Reply:
There is currently a global shared among Widget.cpp and Window.cpp but probably.

Reply:
Unfortunately no, unless each input receives its own TextInputSession instance which is currently not the case, can try to make this cleaner after the fix.